src/binary/uint.rs

```diff
@@ -11,14 +11,14 @@ const MAX_UINT_SIZE_IN_BYTES: usize = mem::size_of::<UIntStorage>();
 /// [UInt and Int Fields](http://amzn.github.io/ion-docs/docs/binary.html#uint-and-int-fields)
 /// section of the binary Ion spec for more details.
 #[derive(Debug)]
-pub struct UInt {
+pub struct DecodedUInt {
```

Comment:
This name is confusing to me--but if you and @zslayton had agreed on this, I am fine with it.

Reply:
Naming is hard. 🙂 I think we need a name pair for `Uint`/`U64Magnitude` that communicates which one was read from a stream (and so is an integer) and which one is being written to a stream (and so is a byte sequence). Open to suggestions.